jellyfin-api/src/main/kotlin/org/jellyfin/sdk/api/client/extensions/MediaInfoApiExtensions.kt

```diff
@@ -0,0 +1,61 @@
+package org.jellyfin.sdk.api.client.extensions
+
+import org.jellyfin.sdk.api.operations.MediaInfoApi
+import org.jellyfin.sdk.model.BitrateMeasurement
+
+/**
+ * Create a [BitrateMeasurement] for a response of [bytes]. The returned value contains the input, duration and the
+ * bitrate.
+ *
+ * @throws AssertionError If the server response is invalid.
+ */
+@Suppress("MagicNumber")
+public suspend fun MediaInfoApi.measureBitrate(bytes: Int): BitrateMeasurement {
+	val now = System.currentTimeMillis()
+
+	// Request the bits
+	getBitrateTestBytes(size = bytes).apply {
+		// Validate response status
+		assert(status == 200) { "Returned status $status is not OK" }
+
+		// Download full content
+		content.awaitContent()
+	}
+
+	// Calculate duration
+	val duration = System.currentTimeMillis() - now
+
+	// Return results
+	return BitrateMeasurement(
+		bytes = bytes,
+		duration = duration,
+		bitrate = bytes / duration * 8000 // bytes -> bits
+	)
+}
+
+/**
+ * Detect the bitrate of the current connection by calling [measureBitrate] until a measurement of at least 1 second or
+ * with the maximum byte size is created.
+ *
+ * @param minBitrate The resulting [BitrateMeasurement.bitrate] field is at least [minBitrate]. Defaults to 1Mb.
+ *
+ * @throws AssertionError If the server response is invalid.
+ */
+@Suppress("MagicNumber")
+public suspend fun MediaInfoApi.detectBitrate(
+	minBitrate: Long = 1_000_000, // 1Mb
+): BitrateMeasurement {
+	val maxBytes = 10_000_000 // Max is 10mb
+	var test = 4_000_000 // start with a response of 4mb
```

Comment:
Would be a better name imo.

```suggestion
	var testBytes = 4_000_000 // start with a response of 4mb
```